apps/examples/mediaplayer/mediaplayer_main.cpp

```diff
@@ -338,12 +369,87 @@ class MediaPlayerController
 };
 
 extern "C" {
+
+static void wifi_sta_connected(wifi_manager_result_e result) {
+	printf("[%s] result %d\n", __FUNCTION__, (int)result);
+}
+
+static void wifi_sta_disconnected(wifi_manager_disconnect_e result) {
+	printf("[%s] result %d\n", __FUNCTION__, (int)result);
+}
+
+static wifi_manager_cb_s wifi_callbacks = {
+	wifi_sta_connected,
+	wifi_sta_disconnected,
+	NULL,
+	NULL,
+	NULL,
+};
+
+static int wifi_connect()
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+	wifi_manager_ap_config_s config;
+
+	/**
+	 * Need to sleep for WiFi initialized.
+	 */
+	int i = 0;
+	for (i = 0; i < 3; i++) {
+		printf("Wait (%d/3) sec...\n", i + 1);
+		sleep(1);
+	}
+
+	printf("wifi_manager_init()...\n");
+	ret = wifi_manager_init(&wifi_callbacks);
+	SAY_OK_OR_ERROR(WIFI_MANAGER_SUCCESS);
+
+	config.ssid_length = strlen(CONFIG_EXAMPLES_MEDIAPLAYER_SSID);
+	config.passphrase_length = strlen(CONFIG_EXAMPLES_MEDIAPLAYER_PASSPHRASE);
+	strncpy(config.ssid, CONFIG_EXAMPLES_MEDIAPLAYER_SSID, config.ssid_length + 1);
+	strncpy(config.passphrase, CONFIG_EXAMPLES_MEDIAPLAYER_PASSPHRASE, config.passphrase_length + 1);
+	config.ap_auth_type = (wifi_manager_ap_auth_type_e)CONFIG_EXAMPLES_MEDIAPLAYER_AUTHENTICATION;
+	config.ap_crypto_type = (wifi_manager_ap_crypto_type_e)CONFIG_EXAMPLES_MEDIAPLAYER_CRYPTO;
+
+	/* current wifi mode is station, then this try will succeed */
+	printf("wifi_manager_connect_ap()...\n");
+	ret = wifi_manager_connect_ap(&config);
+	SAY_OK_OR_ERROR(WIFI_MANAGER_SUCCESS);
+
+	for (i = 0; i < 3; i++) {
+		printf("Wait other (%d/3) sec...\n", i + 1);
+		sleep(1);
+	}
+
+	return 0;
+}
+
+static int wifi_disconnect()
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+	printf("wifi_manager_deinit()...\n");
+	ret = wifi_manager_deinit();
+	SAY_OK_OR_ERROR(WIFI_MANAGER_SUCCESS);
+	return 0;
+}
+
 int mediaplayer_main(int argc, char *argv[])
 {
 	up_cxxinitialize();
 
+	int test = TEST_AAC;
+
+	if (test == TEST_HTTP) {
+		wifi_connect();
+	}
+
```

Comment:
You must wait until wifi_sta_connected is called.
Proceed to next logic after wifi_sta_connected

Reply:
Fixed, at line 451.